crates/algokit_http_client/src/lib.rs

```diff
@@ -18,54 +29,200 @@ pub enum HttpError {
 ///
 /// By default, this trait requires the implementing type to be `Send + Sync`.
 /// For WASM targets, enable the `ffi_wasm` feature to use a different implementation that is compatible with WASM.
-///
-/// With the `ffi_uniffi` feature enabled, this is exported as a foreign trait, meaning it is implemented natively in the foreign language.
-///
 pub trait HttpClient: Send + Sync {
-    async fn get(&self, path: String) -> Result<Vec<u8>, HttpError>;
+    async fn request(
+        &self,
+        method: String,
+        path: String,
+        query: Option<HashMap<String, String>>,
+        body: Option<Vec<u8>>,
+        headers: Option<HashMap<String, String>>,
+    ) -> Result<HttpResponse, HttpError>;
+
+    async fn get(
```

Comment:
There's potentially not need to have get and post if we're exposing a generic request method. I'd either represents all the HTTP methods as individual functions or only expose `request`.

Reply:
Dropped the individual functions, now only uses request

Reply:
Might be worth surfacing this decision on the channel for visibility.